packages/relay/src/lib/eth.ts

```diff
@@ -1305,85 +1305,127 @@ export class EthImpl implements Eth {
       });
   }
 
-  async getLogs(blockHash: string | null, fromBlock: string | null, toBlock: string | null, address: string | null, topics: any[] | null, requestId?: string): Promise<Log[]> {
-    const params: any = {};
-    if (blockHash) {
-      try {
-        const block = await this.mirrorNodeClient.getBlock(blockHash, requestId);
-        if (block) {
-          params.timestamp = [
-            `gte:${block.timestamp.from}`,
-            `lte:${block.timestamp.to}`
-          ];
-        }else {
-          return [];
-        }
+  private async validateBlockHashAndAddTimestampToParams(params: any, blockHash: string, requestId?: string) {
+    try {
+      const block = await this.mirrorNodeClient.getBlock(blockHash, requestId);
+      if (block) {
+        params.timestamp = [
+          `gte:${block.timestamp.from}`,
+          `lte:${block.timestamp.to}`
+        ];
+      } else {
+        return false;
       }
-      catch(e: any) {
-        if (e instanceof MirrorNodeClientError && e.isNotFound()) {
-          return [];
-        }
-
-        throw e;
+    }
+    catch(e: any) {
+      if (e instanceof MirrorNodeClientError && e.isNotFound()) {
+        return false;
       }
+
+      throw e;
+    }
+
+    return true;
+  }
+
+  private async validateBlockRangeAndAddTimestampToParams(params: any, fromBlock: string | null, toBlock: string | null, requestId?: string) {
+    const blockRangeLimit = Number(process.env.ETH_GET_LOGS_BLOCK_RANGE_LIMIT) || constants.DEFAULT_ETH_GET_LOGS_BLOCK_RANGE_LIMIT;
+    if (!fromBlock && !toBlock) {
+      const blockResponse = await this.getHistoricalBlockResponse("latest", true, requestId);
+      params.timestamp = [`gte:${blockResponse.timestamp.from}`, `lte:${blockResponse.timestamp.to}`];
     } else {
-      const blockRangeLimit = Number(process.env.ETH_GET_LOGS_BLOCK_RANGE_LIMIT) || constants.DEFAULT_ETH_GET_LOGS_BLOCK_RANGE_LIMIT;
+      // toBlock is not specified or is `latest` or `pending`
+      if (EthImpl.blockTagIsLatestOrPending(toBlock)) {
```

Comment:
Can't we combine this if with the else if, for example something like `if (!EthImpl.blockTagIsLatestOrPending(toBlock) && toBlockNumber < latestBlockNumber`, imo it is unnecessary to assign "latest" to the `toBlock` if it already is either `latest` or `pending`

Reply:
It could also be `null` in this case.

Reply:
The name `blockTagIsLatestOrPending` does not suggest it checks for null, and personally I don't think it should.

Reply:
Maybe it's better to handle this here, https://github.com/hashgraph/hedera-json-rpc-relay/blob/main/packages/server/src/server.ts#L409-L410 instead of `|| null` to be `|| "latest"`